includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2088,20 +2089,34 @@ protected function request( $params, $api, $method, $is_site_specific = true, $u
 			);
 		}
 
-		$response = $this->http_client->remote_request(
-			[
-				'url'             => $url,
-				'method'          => $method,
-				'headers'         => apply_filters( 'wcpay_api_request_headers', $headers ),
-				'timeout'         => self::API_TIMEOUT_SECONDS,
-				'connect_timeout' => self::API_TIMEOUT_SECONDS,
-			],
-			$body,
-			$is_site_specific,
-			$use_user_token
-		);
+		$headers        = apply_filters( 'wcpay_api_request_headers', $headers );
+		$stop_trying_at = time() + self::API_TIMEOUT_SECONDS;
+		$retries        = 0;
+		$retries_limit  = array_key_exists( 'Idempotency-Key', $headers ) ? self::API_RETRIES_LIMIT : 0;
 
-		$response = apply_filters( 'wcpay_api_request_response', $response, $method, $url, $api );
+		while ( true ) {
+			$response = $this->http_client->remote_request(
+				[
+					'url'             => $url,
+					'method'          => $method,
+					'headers'         => $headers,
+					'timeout'         => self::API_TIMEOUT_SECONDS,
+					'connect_timeout' => self::API_TIMEOUT_SECONDS,
```

Comment:
In relation to the server issue https://github.com/Automattic/woocommerce-payments-server/issues/2682, should we decrease the `connect_timeout` to 10s (the default value in WP) similarly?

Reply:
Same timeout value is applied to requests within admin interface, e.g. retrieving a list of transactions. Such requests can take longer time to be processed by server. However, we can apply different timeouts depending on which method is called in API client class, but I'd do it as a separate change.

Reply:
Just noting that `connect_timeout` is about the connection phase to the server, not the time it takes for the server to process the request.

Quote from the [CURL_CONNECTTIMEOUT doc](https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT.html):
> This only limits the connection phase, it has no impact once it has connected

According to Systems (details can be found in the linked issue), it should take a few milliseconds to connect if everything is healthy, so 70s seems excessive.

The regular `timeout` indeed makes more sense to be greater and it can be complicated to adjust it with relevant values depending on the API targeted, so I was not referring to that.

That can be a separate change, though, as you said 👍 you didn't alter this part

Reply:
> This only limits the connection phase, it has no impact once it has connected

This makes sense, and I agree that it should be a separate issue/PR.